languages/kotlin/generic/Parse_kotlin_tree_sitter.ml

```diff
@@ -31,13 +31,20 @@ module H2 = AST_generic_helpers
 (*****************************************************************************)
 (* Helpers *)
 (*****************************************************************************)
-type env = unit H.env
+
+type context = Program | Pattern
+type env = context H.env
 
 let token = H.token
 let str = H.str
 let fb = Tok.unsafe_fake_bracket
 let sc tok = Tok.sc tok
 
+let in_pattern env =
```

Comment:
I think the usual way we do this is by setting `Flag_semgrep` (unless I misunderstood what's happening here)

Reply:
this is done in the Swift parser: https://github.com/returntocorp/semgrep/blob/78547922c75e15f288f0258a2d4f04e0d60018b8/languages/swift/generic/Parse_swift_tree_sitter.ml#L55
and Julia:
https://github.com/returntocorp/semgrep/blob/78547922c75e15f288f0258a2d4f04e0d60018b8/languages/julia/generic/Parse_julia_tree_sitter.ml#L42

i think in other places we use `sgrep_guard` and the `sgrep_mode` ref, but honestly i think this is cleaner than using a ref. two variants, and two entry points (`pattern` or `program`) are fairly idiomatic for our tree_sitter parsers, i think. what do you think?

Reply:
I don't have a strong opinion either way, just wanted us to be consistent. If we've already started doing this then that's ok with me